array_api_tests/test_creation_functions.py

```diff
@@ -77,9 +78,28 @@ def test_empty(shape, dtype):
         shape = (shape,)
     assert a.shape == shape, "empty() produced an array with an incorrect shape"
 
-# TODO: implement empty_like (requires hypothesis arrays support)
-def test_empty_like():
-    pass
+
+@given(
+    a=xps.arrays(
+        dtype=shared_dtypes,
+        shape=xps.array_shapes(),
+    ),
+    dtype=one_of(none(), shared_dtypes),
```

Comment:
This shouldn't be shared dtypes. The point of the dtype parameter is to specify a different dtype than the input. Actually there's still an open question right now as to whether this should be any set of dtypes or just promotable ones (https://github.com/data-apis/array-api/issues/264). 

For now, we should make `dtypes` be either None or a promotable dtype from `a.dtype`. This will require a new `promotable_dtypes` helper strategy, which would be similar to `mutually_promotable_dtypes` but instead takes a dtype as input and draws a dtype that can be promoted from that dtype in the dtype table (including itself).

Reply:
So there's now a `promotable_dtypes` strategy, being used in the `test_*_like` methods. Note I renamed `mutually_promotable_dtypes` to `mutually_promotable_dtype_pairs` if that's ok (to distinguish what the two strategies generate), and there's a little refactor of the promotion table things in the method `make_dtype_pairs`. Hope it's okay!